### PR TITLE
feat(console): password policy

### PR DESCRIPTION
**Pull request by gao-sun**
> **Note**
> The details like how we display character sets are not finalized yet. this PR should focus on the implementation.

## Summary

follow-up of #4418

- implement the password policy configuration tab in Console
- implement `<NumericInput />` for number inputting

image

## Testing

- test numeric input, try min/max values, try numbers that are out of range
- test various combinations, can save correctly
- empty rows in custom words will be removed automatically

## Checklist

- [ ] `.changeset` (not this PR)
- [ ] ~~unit tests~~
- [ ] integration tests (not this PR)
- [ ] docs (not this PR)

### CODE REVIEW

**Review by wangsijie — APPROVED** [review]

File: packages/console/src/ds-components/TextInput/NumericInput.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  children: React.ReactNode;
+  isDisabled?: boolean;
+};
+
+function Button({ className, onTrigger, children, isDisabled }: ButtonProps) {
+  return (
+    <div
+      role="button"
+      tabIndex={0}
+      className={classNames(styles.button, isDisabled && styles.disabled, className)}
```
Comment (simeng-li) [inline packages/console/src/ds-components/TextInput/NumericInput.tsx:25]: Should we use the <button> tag directly for such use case?

**Comment by simeng-li** [comment]
Need to add the integration tests to cover new added AC features.  P.S. could split the phase update into a separate PR to reduce the PR size.

File: packages/console/src/ds-components/TextInput/NumericInput.tsx (hunk 1 above)
Comment (gao-sun) [reply packages/console/src/ds-components/TextInput/NumericInput.tsx:25]: no it's a `<div>`. i've tried `<button>` but met some strange styling issues

**Comment by gao-sun** [comment]
@simeng-li yeah i'll add tests later because some details are still pending

File: packages/console/src/ds-components/TextInput/NumericInput.tsx (hunk 1 above)
Comment (simeng-li) [reply packages/console/src/ds-components/TextInput/NumericInput.tsx:25]: ok. cc: @charIeszhao there were some places in AC where we have to use div to represent a button element by tweaking lots of a11y code. Maybe we could have a style-cleaned global button css class provided for such a case, so we could use button element directly.

Comment (gao-sun) [reply packages/console/src/ds-components/TextInput/NumericInput.tsx:25]: or we can figure out how to properly style a `<button>`. then no need for these tricky codes

Comment (charIeszhao) [reply packages/console/src/ds-components/TextInput/NumericInput.tsx:25]: Maybe we can simply modify the use of `a11y.ts`, and make it return an object of props, including `role`, `tabIndex`, `onKeyDown` and `onClick`. Then we can spread and merge it into the target element. Thoughts?

**Review by simeng-li — APPROVED** [review]

**Pull request merged**